### ISSUE TITLE
 Delete external authentication (e.g. LDAP / AD, SSO) users from ownCloud, regardles of login success

### ISSUE THREAD

**Issue by MTRichards** [owncloud/core#18542]
As an admin, I want to be able to delete a user from ownCloud that is still valid in LDAP/AD, but that no longer needs access to ownCloud and is no longer in the group filter.

Acceptance Criteria:
- User can be deleted from OCC, provisoning API and the web UI
- User data is deleted no matter the user's state

One possible solution from @blizzz :

> Yes. For instance we could do add an OCC method to flag (and unflag) a user as deleted. Executing this and subsequently calling the user:delete method, this should suffice. With two commands, it's also less likely to do things wrong by accident. But: once it is done, it's done, no undo, except for the backup, you ideally do :speak_no_evil: .

**Comment by MTRichards** [comment]
@cdamken @MorrisJobke

00003201

**Comment by Xenopathic** [comment]
Another possible solution: `user:delete --force`, which attempts to delete the user from the backend but ignores failure to do so (e.g. because it's LDAP) and continues removing all other information.

**Comment by cdamken** [comment]
@MTRichards @blizzz 

> One possible solution from @blizzz :
> 
>     Yes. For instance we could do add an OCC method to flag (and unflag) a user as deleted.

Now that I see this solution, could we probably enhance it for the disable option too? (https://github.com/owncloud/core/issues/12601)

That gives the possibility when the users are not part of the projects that requires ownCloud usage (he/she changes department), but the information could be used for other users (through previous sharing) and the the user joins another project/change department  again where he needs ownCloud again.
Is it possible?

@Xenopathic 
> Another possible solution: user:delete --force, which attempts to delete the user from the backend but

I like that part too, when the previous information should not be kept, but somehow the user can after be flagged again.

**Comment by MTRichards** [comment]
We have discussed a disabled user in ownCloud for quite some time. Folks have asked for it, but it has not been high on the list. If it is simple to add without a lot of work, ok. If it doubles the amount of work, then not really worth stuffing it in, in my opinion.

**Comment by cdamken** [comment]
> Folks have asked for it, but it has not been high on the list. If it is simple to add without a lot of work, ok.

@blizzz How much work will be add 2 flags instead of 1?

**Comment by DeepDiver1975** [comment]
Efford is to be discussed during the planing phase of 9.0. THX

**Comment by johnripper1** [comment]
Just create a group for OC users in the backend (oc_access) and set  the user filter for this group.

**Comment by bboule** [comment]
@cdamken do you have someone waiting on this?

**Comment by cdamken** [comment]
> @cdamken do you have someone waiting on this?

@bboule Yes SFDC 00003201 and most SSO Customers

**Comment by MTRichards** [comment]
But does the workaround fix it?

**Comment by cdamken** [comment]
@MTRichards Which workaround?

**Comment by MTRichards** [comment]
Well, maybe I saw this and thought it was implemented, and it was just a proposal:
>Another possible solution: user:delete --force, which attempts to delete the user from the backend but ignores failure to do so (e.g. because it's LDAP) and continues removing all other information.

Didn't we deploy an SSO delete option for OCC in 8.2.2?
@DeepDiver1975 didn't you do that?

**Comment by cdamken** [comment]
the ```--force``` it's still not implemented

**Comment by DeepDiver1975** [comment]
@MTRichards deleting shib users is implemented in https://github.com/owncloud/user_shibboleth/pull/72 since 8.2.0

**Comment by MTRichards** [comment]
Thanks @DeepDiver1975 that is what I thought, but I also thought the Shib option was for all users at occ, web and proisioning UI, not just SSO users @karlitschek . 
>@bboule Yes SFDC 00003201 and most SSO Customers

Regardless, the above can be shortened to 00003201 only.
The best I can do is 9.1 - and that is so over full the backlog is creaking - unless someone tells me something different.

**Comment by DeepDiver1975** [comment]
> occ, web and proisioning UI

all use the same code which is calling the delete operations on the backends - before that specific fix it was not possible to delete a shib user.

**Comment by PVince81** [comment]
Moving to 9.1

**Comment by MorrisJobke** [comment]
> Moving to 9.1

cc @cdamken

**Comment by PVince81** [comment]
> SSO delete option

?

@DeepDiver1975 @butonic still valid with the user account table ?

cc @pmaier1

**Comment by PVince81** [comment]
An estimate would be nice to see whether there's a quick fix or whether this needs scheduling a bigger block of time.

Moving to backlog for now.

**Comment by PVince81** [comment]
@tomneedham @butonic @jvillafanez is this covered in 10.0.4 / latest user_ldap app and the user:sync process ?

I seem to remember that user:sync asks what to do with the user, whether to delete.

**Comment by PVince81** [comment]
Discussed with the above guys internally, this already works with user:sync command. Closing.
